docs/examples/plot_interval_transposition_error.py

```diff
@@ -0,0 +1,157 @@
+    """
+    idx = irradiance.index
+    # calculate solar position for shifted timestamps:
+    idx = idx + pd.Timedelta(timeshift, unit='min')
+    solpos = location.get_solarposition(idx)
+    # but still report the values with the original timestamps:
+    solpos.index = irradiance.index
+
+    poa_components = pvlib.irradiance.get_total_irradiance(
+        surface_tilt=20,
+        surface_azimuth=180,
+        solar_zenith=solpos['apparent_zenith'],
+        solar_azimuth=solpos['azimuth'],
+        dni=irradiance['dni'],
+        ghi=irradiance['ghi'],
+        dhi=irradiance['dhi'],
+        model='isotropic',
+    )
+    return poa_components['poa_global']
+
+
+# %%
+# Now, calculate the "ground truth" irradiance data.  We'll simulate
+# clear-sky irradiance components at 1-second intervals and calculate
+# the corresponding POA irradiance.  At such a short timescale, the
+# difference between instantaneous and interval-averaged irradiance
+# is negligible.
+
+# baseline: all calculations done at 1-second scale
+location = pvlib.location.Location(40, -80, tz='Etc/GMT+5')
+times = pd.date_range('2019-06-01 05:00', '2019-06-01 19:00',
+                      freq='1s', tz='Etc/GMT+5')
+solpos = location.get_solarposition(times)
+clearsky = location.get_clearsky(times, solar_position=solpos)
+poa_1s = transpose(clearsky, timeshift=0)  # no shift needed for 1s data
+
+# %%
+# Now, we will aggregate the 1-second values into interval averages.
+# To see how the averaging interval affects results, we'll loop over
+# a few common data intervals and accumulate the results.
+
+results = []
+
+for timescale_minutes in [1, 5, 10, 15, 30, 60]:
+
+    timescale_str = f'{timescale_minutes}min'
+    # get the "true" interval average of poa as the baseline for comparison
+    poa_avg = poa_1s.resample(timescale_str).mean()
+    # get interval averages of irradiance components to use for transposition
+    clearsky_avg = clearsky.resample(timescale_str).mean()
+
+    # low-res interval averages of 1-second data, with NO shift
+    poa_avg_noshift = transpose(clearsky_avg, timeshift=0)
+
+    # low-res interval averages of 1-second data, with half-interval shift
+    poa_avg_halfshift = transpose(clearsky_avg, timeshift=timescale_minutes/2)
+
+    df = pd.DataFrame({
+        'ground truth': poa_avg,
+        'modeled, no shift': poa_avg_noshift,
+        'modeled, half shift': poa_avg_halfshift,
+    })
+    # calculate error statistics and save for later
+    error = df.subtract(df['ground truth'], axis=0)
+    stats = error.abs().mean()
+    stats['timescale_minutes'] = timescale_minutes
+    results.append(stats)
+
+df_results = pd.DataFrame(results).set_index('timescale_minutes')
+print(df_results)
+
+# %%
+# The differences shown above are the absolute difference in :math:`W/m^2`.
+# In this example, using the timestamps unadjusted creates an error that
+# increases with increasing interval length, up to a ~40% :math:`W/m^2` error
```

Comment:
I was going to comment that this error depends on time of day, but then I reread the code and saw that you're actually calculating the average error during daylight hours. Somehow need to make that more clear. 

Could also consider restricting the analysis to a couple of times of day. Or add a plot of error as a function of time of day for different shifts (though this depends on the time of year and location...). Maybe this is getting out of scope, sorry.

Reply:
Added a plot along these lines and some other clarifications -- let me know what you think

Reply:
Looks great, thanks!